File: haskell/eval-ex.hs
import Control.Monad
import System.IO
import Text.Printf

fac = product . flip take [1..]
solve x = 1 + sum (map (\t -> x ** fromIntegral t / fac(t)) [1..9]) 

main :: IO ()
main = do
    n_temp <- getLine
    let n = read n_temp :: Int
    forM_ [1..n] $ \a0  -> do
        x_temp <- getLine
        let x = read x_temp :: Double
        printf "%.4f\n" (solve x)  

File: haskell/eval-ex.js
var f = [];
function factorial (n) {
  if (n == 0 || n == 1)
    return 1;
  if (f[n] > 0)
    return f[n];
  return f[n] = factorial(n-1) * n;
}   

solve = x =>     
	Array.from(new Array(9), (_,i) => i + 1)
		.map(t => Math.pow(x, t)/factorial(t))
		.reduce((sum, x) => sum + x, 1);

console.log(solve(20));
